sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -1,112 +1,178 @@
 import scala.concurrent.{Await, Future}
-import scala.util.{Failure, Success}
 import scala.language.postfixOps
-import java.util.{Optional => JOptional}
-import java.math.BigInteger
-import java.util
+import scala.util.{Failure, Success}
 
 
 class EthService(val stateView: AccountStateView, val nodeView: CurrentView[AccountHistory, AccountState, AccountWallet, AccountMemoryPool], val networkParams: NetworkParams, val sidechainSettings: SidechainSettings, val sidechainTransactionActorRef: ActorRef) extends RpcService {
 
-  @RpcMethod("eth_getBlockByNumber") def getBlockByNumber(tag: Quantity, rtnTxObj: Boolean): EthBlock.Block = { //TODO: Implement getting block information
-    new EthBlock.Block("0x1", "0x56", "0x57", "0x58", "0x59", "0x0", "0x0", "0x0", "0x0", "0", "0", "0", "0", "0", "0", "1", "3000", "2000", "22", new util.ArrayList[EthBlock.TransactionResult[_]], new util.ArrayList[String], new util.ArrayList[String], "")
-  }
-
-
-  @RpcMethod("eth_call") def call(ethCall: util.LinkedHashMap[String, String], tag: Quantity): Data = { // Executes a new message call immediately without creating a transaction on the block chain
-    // TODO: add read-only evm execution and cleanup here
-    var processor = new EvmMessageProcessor();
-    val msg = new Message(
-      new AddressProposition(Numeric.hexStringToByteArray("0xA83035b4E719827f146C91b119508ea452c77F35")),
-      new AddressProposition(Numeric.hexStringToByteArray(ethCall.get("to"))),
-      BigInteger.ZERO,
-      BigInteger.ZERO,
-      BigInteger.ZERO,
-      BigInteger.ZERO,
-      BigInteger.ZERO,
-      BigInteger.ZERO,
-      Numeric.hexStringToByteArray(ethCall.get("data")))
-    if (processor.canProcess(msg, stateView)) {
-      var result = processor.process(msg, stateView)
-      if (!result.isFailed)
-        return new Data(result.asInstanceOf[ExecutionSucceeded].returnData())
+  //function which describes default transaction representation for answer after adding the transaction to a memory pool
+  val defaultTransactionResponseRepresentation: (Transaction => SuccessResponse) = {
+    transaction => TransactionIdDTO(transaction.id)
+  }
+
+  @RpcMethod("eth_getBlockByNumber") def getBlockByNumber(tag: Quantity, hydratedTx: Boolean): EthereumBlock = {
+    val blockId = nodeView.history.getBlockIdByHeight(Integer.parseInt(Numeric.cleanHexPrefix(tag.getValue), 16))
+    val txHashes: ListBuffer[String] = ListBuffer()
+    val txViews: ListBuffer[EthereumTransactionView] = ListBuffer()
+
+    if (blockId.isEmpty) return new EthereumBlock()
+
+    val txs = getTransactionsFromBlock(getBlockById(blockId.get()))
+    txs.foreach(transaction => {
+      txViews.append(new EthereumTransactionView(transaction))
+      txHashes.append(Numeric.prependHexPrefix(transaction.id))
+    })
+    new EthereumBlock(tag.getValue, Numeric.prependHexPrefix(blockId.get()), txHashes.toList.asJava, txViews.toList.asJava, hydratedTx, nodeView.history.getBlockById(blockId.get()).get())
```

Comment:
`nodeView.history.getBlockById(blockId.get()).get()`
don't need to get block again from the storage - it costs.
we already have a block in line 53: `val txs = getTransactionsFromBlock(getBlockById(blockId.get()))`
let's assign it to some val and reuse here.